src/main/java/teammates/sqllogic/core/DeadlineExtensionsLogic.java

```diff
@@ -37,14 +37,27 @@ void initLogicDependencies(DeadlineExtensionsDb deadlineExtensionsDb) {
      * Get extended deadline for this session and user if it exists, otherwise get the deadline of the session.
      */
     public Instant getDeadlineForUser(FeedbackSession session, User user) {
-        DeadlineExtension deadlineExtension =
```

Comment:
This is another discovered bug, the session and user id params are flipped, hence it will never be able to find any DeadlineExtensions.